src/lib/OS/MemRegion.cc

```diff
@@ -123,10 +118,10 @@ uint64_t MemRegion::addVma(VMA* vma, uint64_t startAddr) {
   // same. Here startAddr is either mmap pointer or an address greater than mmap
   // pointer.
   if (!allocated) {
-    startAddr = startAddr >= mmapPtr_ ? startAddr : mmapPtr_;
+    startAddr = startAddr >= mmapReg_->mmapPtr ? startAddr : mmapReg_->mmapPtr;
```

Comment:
Could you add brackets around `startAddr >= mmapReg_->mmapPtr` for readability please